src/sicmutils/value.cljc

```diff
@@ -49,23 +49,37 @@
 #?(:cljs
    (def ^:private ratio? (constantly false)))
 
-(extend-type #?(:clj Object :cljs default)
-  Value
-  (nullity? [o] (and (number? o) (core-zero? o)))
+(extend-protocol Value
+  #?(:clj Number :cljs number)
+  (nullity? [x] (core-zero? x))
+  (unity? [x] (== 1 x))
+  (zero-like [_] 0)
+  (one-like [_] 1)
+  (freeze [x] x)
+  (exact? [x] (or (integer? x) (ratio? x)))
+  (numerical? [_] true)
+  (kind [x] (type x))
+
+  nil
+  (freeze [_] nil)
+  (numerical? [_] false)
+  (nullity? [_] true)
+  (unity?[_] false)
+  (kind [_] nil)
+
+  #?(:clj Object :cljs default)
+  (nullity? [o] false)
   (numerical? [_] false)
-  (unity? [o] (and (number? o) (== o 1)))
-  (exact? [o] (or (integer? o) (ratio? o)))
-  (zero-like [o] (cond (number? o) 0
-                       (instance? Symbol o) 0
+  (unity? [o] false)
```

Comment:
base implementation gets simpler, since we're explicitly handling these cases in the number implementation above.